server/src/resolvers/Mutation.js

```diff
@@ -0,0 +1,203 @@
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+const { APP_SECRET, STATUS, isBuddyAuth } = require("../utils");
+
+async function addBuddy(parent, args, context) {
+  await isBuddyAuth(context)
```

Comment:
you are repeating this in every query, isn't anything in GraphQL like middleware in express?

Reply:
That's a good catch, definitely there should be some middleware 👍